src/Downloader.php

```diff
@@ -185,7 +185,7 @@ private function switchToDegradedMode(\Exception $e, string $url): void
         $this->degradedMode = true;
     }
 
-    private function getOptions(array $headers): array
+    private function getOptions(iterable $headers): array
```

Comment:
this looks wrong to me. Passing an iterator will not be allowed. The headers inside the options of the Composer RemoteFilesystem must be an array, not any iterable